Selenium_pract_task/Tets_Cases/API_Test_Cases/Users_Endpoint_Tests .cs

```diff
@@ -0,0 +1,125 @@
+
+            Assert.That(contentHeaders, Does.Contain("application/json; charset=utf-8"),
+                $"Content-Type does not contain 'application/json'. Actual: {contentHeaders}");
+
+            logger.Information("ValidateContentTypeHeaderTest passed successfully.");
+        }
+
+        [Test]
+        public void ValidateUserCreationTest()
+        {
+            logger.Information("Starting ValidateUserCreationTest.");
+
+            var response = _usersEndpoint.CreateUser("John Doe", "johndoe");
+
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.Created),
+                "Expected status code 201.");
+
+            var responseBody = _usersEndpoint.DeserializeUserCreationResponse(response);
+            Assert.That(responseBody.id, Is.Not.Null,
+                "Response does not contain ID field.");
+
+            logger.Information("ValidateUserCreationTest passed successfully.");
+        }
+
+        [Test]
+        public void ValidateUserNotifiedIfResourceNotExist()
+        {
+            logger.Information("Starting ValidateUserNotifiedIfResourceNotExist.");
+
+            var invalidEndpoint = new UserService($"{userApiEndpoint}/invalidendpoint");
+            var response = invalidEndpoint.GetUsersResponse();
+
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.NotFound),
+                "Expected status code 404.");
+
+            logger.Information("ValidateUserNotifiedIfResourceNotExist passed successfully.");
+        }
+
+        [Test]
+        public void ValidateUsersResponseBodyTest()
+        {
+            logger.Information("Starting ValidateUsersListTest.");
+
+            var response = _usersEndpoint.GetUsersResponse();
+
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK), $"Expected status code 200 but was: [{response.StatusCode}]");
+
+            var users = _usersEndpoint.DeserializeUsers(response);
+
+            Assert.That(users, Is.Not.Null, "Users list is null.");
+            Assert.That(users.Count, Is.EqualTo(10), "Expected 10 users in the response.");
+
+            foreach (var user in users)
+            {
+                Assert.That(user.Name, Is.Not.Null, "User name is null.");
+                Assert.That(user.Username, Is.Not.Null, "User username is null.");
+                Assert.That(user.Company, Is.Not.Null, "User company is null.");
+            }
+
+            logger.Information("ValidateUsersListTest passed successfully.");
+        }
+
+        [Test]
+        public void ValidateUsersListTest()
+        {
+            logger.Information("Starting ValidateUsersResponseBodyTest.");
+
+            var response = _usersEndpoint.GetUsersResponse();
+
+            Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.OK), $"Expected status code 200 but was: [{response.StatusCode}]");
+
+            var users = _usersEndpoint.DeserializeUsers(response);
+
+            foreach (var user in users)
```

Comment:
Validate that user recives a list of users with the following information: "id", "name", "username", "email", "address”, "phone", "website", "company";  
Just check JSON Shema instead of checking every field